3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,61 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+const BASE_URL = 'https://pokeapi.co/api/v2/pokemon/';
+
+async function fetchData(url) {
+    const response = await fetch(url);
+    if (!response.ok) {
+        throw new Error('Failed to fetch');
+    }
+    return response.json();
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+    const data = await fetchData(`${BASE_URL}?offset=151&limit=151`);
+    const pokemons = data.results;
+    const selectElement = document.createElement('select');
+    selectElement.addEventListener('change', (e) => { fetchImage(e.target.value) });
+    document.body.appendChild(selectElement);
+    pokemons.forEach((pokemon) => {
+        const optionElement = document.createElement('option');
+        optionElement.textContent = pokemon.name;
+        optionElement.value = pokemon.name;
+        selectElement.appendChild(optionElement);
+    })
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+async function fetchImage(pokemonName) {
+    try {
+
+        const imgUrl = `${BASE_URL}${pokemonName}`;
+        const response = await fetch(imgUrl);
```

Comment:
When you are using a `try ... catch` block, look at what can throw errors and wrap around this. You don't need to include everything in this block. Also, why are you not using your `fetchData`?

```suggestion
try {
        const response = await fetch(imgUrl);
} catch (error) etcetc
```

Reply:
Thanks. will checkout it out and push refactored codes.